docs/Inspection-Concept.md

```diff
@@ -0,0 +1,80 @@
+# Machinery inspection concept
+
+This document describes the concept of inspecting systems in Machinery. System
+inspection is one of the central features of Machinery. It gathers all data
+about a given system and stores it in a description. The description can then be
+used for a variety of use cases, including display, archive, export to other
+formats, and more. Inspection is implemented as the `machinery inspect` command.
+
+## Access to inspected system
+
+To gather data Machinery executes a number of commands on the inspected system,
+collects and processes the results, and stores them in the format of a system
+description on the local system, where Machinery is executed.
+
+Usually the inspected system will be different from the system where Machinery
+is running. Machinery will use ssh to access the remote system. For the special
+case where the inspected system is the local system where Machinery is run, the
+same commands will be executed locaclly, not via ssh.
+
+## Privileges
+
+Machinery is usually run by a regular user. A remote inspection does not
+require root rights on the system where Machinery is run.
+
+On the inspected system root rights are required for some commands which access
+data, which is not accessible by a regular user.
+
+Machinery supports two modes how to execute commands on a remote system with
+root privileges. The default mode is to log in as root via ssh. So the root
+account needs to be configured to be accessed by the user who runs Machinery.
+This is done by copying the public ssh key of the Machinery user to the root
+account of the inspected system. Machinery requires the Machinery user to use
+ssh-agent to allow password-less login to the remote machine.
```

Comment:
Theoretically password-less login works without `ssh-agent` if you use keys without passphrase.